src/backend/Csrs.Api/Services/MessageService.cs

```diff
@@ -51,10 +51,19 @@ public async Task<IList<Message>> GetPartyMessages(string partyId, CancellationT
 
         public async Task SetMessageRead(string messageGuid, CancellationToken cancellationToken)
         {
-
             _logger.LogDebug("Set party message read request recieved");
 
-            throw new NotImplementedException();
+            var select = new List<string>() {"ssg_csrsmessageread"};
+
+            var communicationMessage = await _dynamicsClient.Ssgcsrscommunicationmessages.GetByKeyAsync(messageGuid, select, null, cancellationToken);
+            if (communicationMessage is null) throw new ArgumentNullException(messageGuid, "Invalid message Guid");
```

Comment:
If the API is unable to complete a job because it is missing requires stuff or doesnt get a result throw exception and Log a message so bad stuff isnt happening and no audit trail. 
EX) 
_logger.LogInformation("No associated party, cannot create User Request");
                    throw new HttpOperationException("No Party Associated");

Reply:
Got it thanks!